crates/ruff_python_parser/src/error.rs

```diff
@@ -536,6 +536,7 @@ pub enum UnsupportedSyntaxErrorKind {
     TypeParameterList,
     TypeAliasStatement,
     TypeParamDefault,
+    ParenthesizedContextManager,
```

Comment:
It would be nice to add documentation, similar to what you did for other variants.